modules/auxiliary/admin/mysql/mysql_enum.rb

```diff
@@ -6,6 +6,7 @@
 class MetasploitModule < Msf::Auxiliary
   include Msf::Auxiliary::Report
   include Msf::Exploit::Remote::MYSQL
+  include Msf::OptionalSession
```

Comment:
If we're moving to using optional sessions like this here:

```suggestion
  include Msf::OptionalSession::MySQL
```

Why wouldn't we just override `rport` and `rhost` to be something like:

```ruby
def rhost
  return mysql_conn.host if mysql_conn.host
  super # Or potentially default to datastore['RHOST'] etc
end

def rport
  return mysql_conn.host if mysql_conn.host
  super
end
```

That way none of these modules would need code changes to replace rhost->mysql_conn.host

Reply:
I was thinking directly accessing the host/port via the client makes the most sense here, it feels a bit weird to me trying to use `rhost` and `rport` when we require the client in every situation anyway, I'm just not sure that getting a result for `rhost` when the datastore value is empty is expected either

I'm not _super_ against it though, unless there's a scenario where we have an OptionalSession _and_ want to keep rhost/rport datastore options free but that feels like a bit of a stretch

any references to `datastore['RHOST']` or `datastore['RPORT']` still need to be changed either way whether it's to `rhost`/`rport` or `mysql_conn.host`

Reply:
Works for me :+1: It also stops mixin madness, for modules that would need to target multiple protocols